### PR TITLE
Handle video error

### PR DESCRIPTION
**Pull request by zzarcon**

### CODE REVIEW

File: example/app.tsx [hunk 1]
```diff
@@ -43,11 +43,17 @@ export default class App extends Component<AppProps, AppState> {
...
     const videoUrl = URL.createObjectURL(this.videoFile);
     this.snapshoter = new VideoSnapshot(this.videoFile!);
-    const videoPreview = await this.snapshoter.takeSnapshot();
+    try {
+      const videoPreview = await this.snapshoter.takeSnapshot();
+      this.setState({
+        videoPreview
+      });
+    } catch (e) {
+      console.log(e)
```
Comment (soswow) [inline example/app.tsx:52]: prob. `console.error` ?

File: src/index.ts [hunk 2]
```diff
@@ -72,6 +71,9 @@ class VideoSnapshot {
         video.pause();
         resolve(video);
       });
+      video.addEventListener('error', () => {
+        reject('failed to load video');
```
Comment (soswow) [inline src/index.ts:75]: will there be an argument for that callback? (error itself)
Also, it is not cool to throw strings as an error. how about we create an error object, maybe attach original one somehow if we get anything here as an argument?

Comment (zzarcon) [reply src/index.ts:75]: the argument is the event, which contains no useful information about the error. I will create an Error and throw it, that's a good idea

**Comment by soswow** [comment]
LGTM

**Pull request merged**